Simplify "Installation" documentation

lawcho opened the pull request:
See commit message for explanation of changes.

I built this locally in `nix develop` with `make user-manual-html`, which showed no warnings from the changed files.

lawcho commented:
Despite the GitHub actions outage, CI built the docs. A preview is at:

https://agda--8089.org.readthedocs.build/en/8089/getting-started/installation.html

L-TChen commented:
Using `Repology.org` is nice! But, I was overwhelmed by the table of package managers and I couldn't find `homebrew` at first sight. Can we just use tiny badges for major package managers on macOS, Ubuntu with a link to https://repology.org/project/agda/badges?

lawcho commented:
Thanks for the feedback!

> I couldn't find homebrew at first sight

If homebrew shipped Agda 2.8.0, repology would highlight it green.
Would that be enough emphasis?

> Can we just use tiny badges for major package managers

I'm reluctant to replace the big table with tiny badges because:

* It creates extra maintenance effort: tiny-badge URLs need to be updated whenever a new Ubuntu release is made, a new package manager becomes 'major', etc.
* Highlighting the up-to-date distributions is more valuable to prospective users than highlighting the 'major' distributions
* An "update your distribution to get highlighted" policy encourages distribution maintainers to ship the latest Agda

L-TChen commented:
> I'm reluctant to replace the big table with tiny badges because:
> 
> * It creates extra maintenance effort: tiny-badge URLs need to be updated whenever a new Ubuntu release is made, a new package manager becomes 'major', etc.
> * Highlighting the up-to-date distributions is more valuable to prospective users than highlighting the 'major' distributions
> * An "update your distribution to get highlighted" policy encourages distribution maintainers to ship the latest Agda

How about excluding unsupported platforms and other sources that are not really binary package managers? I still feel the information provided by this table is overwhelming for a beginner, but it might be better than the current one.
![](https://repology.org/badge/vertical-allrepos/agda.svg?exclude_sources=site,modules&columns=4&exclude_unsupported=1)

lawcho committed:
Tune table for search

lawcho commented:
The table in [the preview](https://agda--8089.org.readthedocs.build/en/8089/getting-started/installation.html#option-1-from-package-manager) is now shorter & searchable with a web browser's Ctrl+F.

lawcho committed:
Recommend binaries over distro package managers

lawcho committed:
Fix typo

L-TChen commented:
(Sorry for not getting back to this PR sooner!)

andreasabel commented on doc/user-manual/getting-started/installation.rst:
```diff
@@ -18,28 +18,10 @@ Step 1: Install Agda
 .. _prebuilt-agda-from-github:
 
-Option 2: From GitHub releases
+Option : From GitHub releases
```
I suppose this was meant to be `Option 1`.

jespercockx commented on HACKING.md:
```diff
@@ -31,6 +31,182 @@ Note: Some instructions in this document are likely outdated,
+How to build Agda from source
+=============================
+
+With nix
```
Since most developers do not use Nix, I think it would be more reasonable to start this with the "without nix" section.

jespercockx commented:
@lawcho this looks almost good to go. Could you please fix the conflicts that arose and address the remaining comments? Then I'll merge it.

lawcho committed:
Simplify "Installation" documentation

The page is now much shorter, and written exclusively for first-time Agda users.
Troubleshooting, customization, and special-case documentation is kept out of this page.

Text moved:

* Installation via package-manager to 1st option
  * Because distro packagers know their distro better than us,
    so their Agda is more likely to work than Cabal's
* Emacs-mode details to Emacs page
  * Because these are uninteresting to non-Emacs users
* Cabal tips to HACKING.md
  * "zlib and ncurses Dependency"
  * "Install alex and happy dependencies"
  * "Installing Multiple Versions of Agda"
  * "Keeping the Default Environment Clean"
  * "Installing ICU"
  * "Installation Flags"
  * "Cabal install fails due to dynamic linking issues"
* Installation-troubleshooting tips to new page
  * Admonition RE macOS notarisation
  * "A Common Issue on Windows: Invalid Byte Sequence"
* Optional steps to own sections at end
  * "Install the Agda Standard Library"
  * Admonition RE GHC installation

Text deleted:

* Section "A Common Issue: Missing ieee754 Dependency"
  * Failure does not occur at install-time
  * Redundant with "Keeping the Default Environment Clean"
* Admonition RE `cluster-counting`
  * AFAICT this is only needed for `--count-clusters`
  * But this restriction is already well documented
    (e.g. in `--count-clusters` error message & help)
* Section "Prebuilt Packages and System-Specific Instructions"
  * Avaiability information duplicated from repology
  * Instuctions to report bugs to package maintainers
    risked mis-directing reports of Agda bugs
  * Other information probably already known by users
    of those package managers (e.g. how to call `apt`)
* Section "Tested GHC Versions"
  * Duplicated from Agda.cabal, developer-only
* Misc. redundant verbiage (e.g. in headers, introduction)

Additions:

* Repology-generated badge of packaging status
* Mention installation via VS Code mode
* Mention nix alongside cabal/stack
* Mention JS backend alongside MAlonzo
* Section intros to new parts of HACKING.md

lawcho committed:
Move nix build instructions after cabal

lawcho commented:
I force-pushed from 82d0e60 to 6f7be5d to resolve conflicts & implement requested changes.

lawcho committed:
Patch for CI

jespercockx commented:
Thank you for the updates! Please run `make fix-whitespace` and `make user-manual-html` and resolve the errors to make sure CI is green.

jespercockx commented:
Thank you very much for your work on this PR! I'll merge this now.
